### DISCUSSION TITLE
How the current verification was overriden

### DISCUSSION THREAD

**Discussion by MahtabAlam991** [ttdyce/NHentai-NHViewer#28] [Q&A] [answered]
I stumbled upon this and what I found fascinating at least for me was that this app was able to override the human verification that the official website imposes, I would love to know how this was achieved.

**Comment by ttdyce** [comment, marked as answer]
Actually, sometimes the site skip the human verification automatically and the app just check if it happens. 
It is done using a hidden webview to open the site, and wait for any cookie returned: got cookie means skipped verify, no cookie means required verify. 

You can see the code around here: 

https://github.com/ttdyce/NHentai-NHViewer/blob/3bc7e4c4fa0d03ddbc662e402b11e25d0cb3dd5a/app/src/main/java/com/github/ttdyce/nhviewer/view/SplashActivity.java#L49-L69

https://github.com/ttdyce/NHentai-NHViewer/blob/3bc7e4c4fa0d03ddbc662e402b11e25d0cb3dd5a/app/src/main/java/com/github/ttdyce/nhviewer/view/RefreshCookieActivity.java#L16-L70

**Reply by MahtabAlam991** [reply]
Thank you.....
